includes/rest/class-inbox-controller.php

```diff
@@ -127,27 +128,31 @@ public function register_routes() {
 	public function create_item( $request ) {
 		$data     = $request->get_json_params();
 		$activity = Activity::init_from_array( $data );
-		$type     = $request->get_param( 'type' );
-		$type     = \strtolower( $type );
-
-		/**
-		 * ActivityPub inbox action.
-		 *
-		 * @param array              $data     The data array.
-		 * @param int|null           $user_id  The user ID.
-		 * @param string             $type     The type of the activity.
-		 * @param Activity|\WP_Error $activity The Activity object.
-		 */
-		\do_action( 'activitypub_inbox', $data, null, $type, $activity );
-
-		/**
-		 * ActivityPub inbox action for specific activity types.
-		 *
-		 * @param array              $data     The data array.
-		 * @param int|null           $user_id  The user ID.
-		 * @param Activity|\WP_Error $activity The Activity object.
-		 */
-		\do_action( 'activitypub_inbox_' . $type, $data, null, $activity );
+		$type     = \strtolower( $request->get_param( 'type' ) );
+
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+		if ( wp_check_comment_disallowed_list( $activity->to_json( false ), '', '', '', $_SERVER['REMOTE_ADDR'], $_SERVER['HTTP_USER_AGENT'] ?? '' ) ) {
+			Debug::write_log( 'Blocked activity from: ' . $activity->get_actor() );
```

Comment:
Let's prepare the responses in the if and else and prepare a forbidden error in this case.

Maybe also: https://codeberg.org/fediverse/fep/src/branch/main/fep/c180/fep-c180.md#actor-not-authorized (but this can be done as part of #1563 )

Reply:
I tried to find out whether there were expectations around response codes etc when activities from a blocked user make it to an inbox but couldn't really find much.

My understanding of the desired behavior is that any blocked user shouldn't be notified/see that they're blocked, so returning the general success message while we deal with the content internally seemed prudent to me.

The request itself was successful after all, we just decide to not do anything with the contents.

Reply:
An API error is not user-facing and might help the API to limit requests, but I am fine with the PR as a first step.